server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -941,7 +941,13 @@ public Info(String name, ThreadPoolType type, int min, int max, @Nullable TimeVa
 
         public Info(StreamInput in) throws IOException {
             name = in.readString();
-            type = ThreadPoolType.fromType(in.readString());
+            String typeAsString = in.readString();
+            if ("fixed_auto_queue_size".equals(typeAsString) || "direct".equals(typeAsString)) {
```

Comment:
The enum definitions are still there, so could we just `switch` on the enum?

Reply:
++
Or, we remove the enum definitions too :)

Reply:
Replaced with switch